doc/Guide to Contributing to Sia.md

```diff
@@ -0,0 +1,183 @@
+
+You should install the latest [official Go binary][binary] for your system (if not available, [install from source][source]).
+If you plan to cross compile Sia, see [Cross Compilation with Go 1.5][cross] by Dave Cheney.  
+
+Now make a workspace directory in which you will store source code and dependencies.  You can choose any filepath except where you installed Go (don't choose `/usr/local`).
+```bash
+# make a working directory called golang in your home directory
+$ mkdir $HOME/golang
+# store base path in an environmental variable
+$ echo 'export GOROOT=$HOME/golang' >> $HOME/.profile
+# add bin subdirectory to PATH environmental variable
+$ echo 'export PATH=$PATH:$GOROOT/bin' >> $HOME/.profile
+```
+Make a base path for your code.  (You'll need [a Github account][signup].)
+```
+$ mkdir -p $GOPATH/src/GitHub.com/<your GitHub username>
+```
+
+<a name="learn-go"/>
+### Learn Go
+* To get familiar with the language, start with the official [Tour of Go][tour].
+* Move onto [How to Write Go Code][how] to learn how to organize Go packages and use the go tool.
+* Finish with the [Effective Go][effective] guide.
+
+<a name="build-sia"/>
+## Build Sia
+To build Sia on your machine, enter the following on the command line:
+```bash
+# Download Sia and its dependencies
+# Binaries will be installed in $GOROOT/bin
+$ go get -u GitHub.com/NebulousLabs/Sia/...
+
+# Switch to directory containing Sia source code
+$ cd $GOROOT/src/GitHub.com/NebulousLabs/Sia
+
+# You have three Sia builds to choose from.
+# To build the standard release binary:
+$ make release-std
+# Or to build the release binary with race detection and an array debugging asserts:
+$ make release
+# Or to build the developer binary (with a different genesis block, faster block times, and other changes):
+$ make
+```
+
+<a name="contribute"/>
+## Contribute to the codebase
+<a name="setup-git"/>
+### Set up git
+Install git on your machine according to [these instructions][install-git] in the Pro Git book.
+
+You will first need to set up global settings using the command line.
+```bash
+$ git config --global user.name "Your Name"
+$ git config --global user.email you@somedomain.com
+
+# Tell git to remember your login information for a certain amount of time.
+# Default time is 15 minutes:
+$ git config --global credential.helper cache
+# Or you can choose a different amount of time:
+$ git config --global credential.helper "cache --timeout=[seconds]"
+
+```
+<a name="fork"/>
+### Fork the Sia repository
+While logged into your Github account, navigate to the [Sia repository][sia] and click the 'Fork' button in the upper righthand corner.  Your account now has a 'forked' copy of the original repo at `https://GitHub.com/<your GitHub username>/Sia`.
+
+Clone this forked repo into a local directory:
+```bash
+$ cd $GOPATH/src/GitHub.com/<your GitHub username>
+$ git clone https://GitHub.com/<your GitHub username>/Sia
```

Comment:
I think this is not the preferred way of contributing to Go projects, because it screws up your imports. That is, the `go` tool will complain if the package import is `"github.com/NebulousLabs/Sia"` but it's in a local directory called `$GOPATH/src/github.com/lukechampine/Sia`.

Instead, the process should be: fork the repo, cd to your `NebulousLabs/Sia` directory, and add your fork as a remote.